core/src/engine/task/work_stealing_queue/consumer.hpp

```diff
@@ -33,29 +36,29 @@ class Consumer final {
   void SetIndex(std::size_t index) noexcept;
   bool IsStopped() const noexcept;
   void MoveTasksToGlobalQueue(impl::TaskContext* extra);
-  impl::TaskContext* StealFromAnotherConsumer(std::size_t attempts,
+  impl::TaskContext* StealFromAnotherConsumer(const std::size_t attempts,
                                               std::size_t to_steal);
   std::size_t Steal(utils::span<impl::TaskContext*> buffer);
-  impl::TaskContext* TryPopFromGlobalQueue();
+  impl::TaskContext* TryPopFromOwnerQueue(bool is_global);
+  impl::TaskContext* ProbabilisticPopFromOwnerQueues();
   impl::TaskContext* TryPop();
+  impl::TaskContext* TryPopBeforeSleep();
   impl::TaskContext* DoPop();
-  void Sleep(std::int32_t val);
+  void Sleep(const std::int32_t old_sleep_counter);
   void WakeUp();
 
-  LocalQueue<impl::TaskContext, 63> local_queue_;
-  WorkStealingTaskQueue* const owner_;
-  ConsumersManager* const consumers_manager_;
+  LocalQueue<impl::TaskContext, 63> local_queue_{};
```

Comment:
Extract magical constants into named constants, place them at the beginning of `private` section, after type declarations

Explain in a comment the logic behind `63`